src/utils/wallet.js

```diff
@@ -211,6 +215,74 @@ const wallet = {
     }
   },
 
+  /**
+   * The wallet needs each token metadata to show information correctly
+   * So we fetch the tokens metadata and store on redux
+   *
+   * @param {Array} tokens Array of token uids
+   * @param {String} network Network name
+   * @param {Number} downloadRetry Number of retries already done
+   *
+   * @memberof Wallet
+   * @inner
+   **/
+  async fetchTokensMetadata(tokens, network, downloadRetry = 0) {
+    const metadataPerToken = {};
+    let didBreak = false;
+
+    const handleRetry = (index) => {
+      if (downloadRetry === METADATA_RETRY_LIMIT) {
+        // Should we throw an error and prevent wallet usage only because metadata was not loaded correctly?
+        // Right now I'm continuing and will download as many as we can
+        // Won't break the for
+        return false;
+      } else {
+        setTimeout(() => {
+          this.fetchTokensMetadata(tokens.slice(index), network, downloadRetry + 1);
+        }, DOWNLOAD_METADATA_RETRY_INTERVAL);
+        // Will break the for
+        didBreak = true;
+        return true;
+      }
+    }
+
+    for (const [index, token] of tokens.entries()) {
+      try {
+        const response = await metadataApi.getDagMetadata(token, network);
```

Comment:
I feel that this retry logic should be inside `getDagMetadata(...)` which might receive an optional parameter `maxNumberOfRetries`. It should even handle the issue of getting a response with no data.

Reply:
You can move the current implementation of `getDagMetadata(...)` to `_getRawDagMetadata(...)` so we can use it for a single fetch (even though I feel we won't).

Reply:
Done! Thanks!